pkg/controller/sync/placement/resource.go

```diff
@@ -32,21 +32,19 @@ type ResourcePlacementPlugin struct {
 	// Informer controller for placement directives of the federated type
 	controller cache.Controller
 
-	// Whether to default to all clusters if a placement resource is
-	// not found for a given qualified name.
-	defaultAll bool
+	kind string
 }
 
-func NewResourcePlacementPlugin(client util.ResourceClient, targetNamespace string, triggerFunc func(pkgruntime.Object), defaultAll bool) PlacementPlugin {
-	return newResourcePlacementPluginWithOk(client, targetNamespace, triggerFunc, defaultAll)
+func NewResourcePlacementPlugin(client util.ResourceClient, targetNamespace string, triggerFunc func(pkgruntime.Object)) PlacementPlugin {
+	return newResourcePlacementPluginWithOk(client, targetNamespace, triggerFunc)
 }
 
-func newResourcePlacementPluginWithOk(client util.ResourceClient, targetNamespace string, triggerFunc func(pkgruntime.Object), defaultAll bool) *ResourcePlacementPlugin {
+func newResourcePlacementPluginWithOk(client util.ResourceClient, targetNamespace string, triggerFunc func(pkgruntime.Object)) *ResourcePlacementPlugin {
```

Comment:
"WithOK" seems tell nothing. What about remove it or use Impl as postfix? Same as computePlacementWithOK()

Reply:
Please limit yourself to commenting on changes rather than existing code.  `withOK` is existing code.

If you examine the definition of `ResourcePlacementPlugin` you will notice that it has a method `computePlacementWithOk` that returns a boolean indication of whether placement could be successfully retrieved.  If you examine the definition of `NamespacedPlacementPlugin`, you will find that this method is used to determine whether namespace placement was retrieved, and if so, the intersection of namespace and resource placement is used.  If namespace placement was not retrieved, the resource placement is used by itself.

Also, please avoid reviewing PRs that clearly state 'this PR is not ready for review' (and/or WIP in the title).  If a feature requires more than 1 PR to implement, I will frequently post them in a chain with subsequent PRs marked WIP until their predecessor has merged.  This avoids reviewing follow-on PRs until all review-related changes from previous PRs have been incorporated.

Reply:
Got it. I will review it after it is ready for review.